modules/sfp_stackoverflow.py

```diff
@@ -0,0 +1,255 @@
+
+    def extractIP6s(self, text):
+        ips = set()
+
+        matches = re.findall(r'(?:^|(?<=\s))(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))(?=\s|$)', text)
+
+        if matches:
+            for match in matches:
+                if self.sf.validIP6(match) and not(self.sf.isValidLocalOrLoopbackIP(match)):
+                    ips.add(match)
+            return list(ips)
+        else:
+            return None
+
+    def handleEvent(self, event):
+        eventData = event.data
+
+        if self.errorState:
+            return
+
+        if eventData in self.results:
+            self.debug(f"Skipping {eventData}, already checked.")
+            return
+
+        self.results[eventData] = True
+
+        query_results = self.query(eventData, "excerpts")
+        items = query_results.get('items')
+        allEmails = []
+        allUsernames = []
+        allIP4s = []
+        allIP6s = []
+
+        if items is None:
+            return
+        # Iterate through all results from query, creating raw_rir_data events and extracting emails
+        for item in items:
+            if self.checkForStop():
+                return
+
+            body = item["body"]
+            excerpt = item["excerpt"]
+            question = item["question_id"]
+            text = body + excerpt
+
+            # create raw_rir_data event
+            e = SpiderFootEvent('RAW_RIR_DATA',
+                                str("<SFURL>https://stackoverflow.com/questions/") + str(question) + str("</SFURL>") + str("\n") + str(item), self.__name__, event)
+            self.notifyListeners(e)
+
+            emails = self.sf.parseEmails(text)
+            if emails is not None:
+                for email in emails:
+                    allEmails.append(str(email))
+
+            questionId = item["question_id"]
+            username = self.extractUsername(questionId, event)
+            if username is not None:
+                allUsernames.append(username)
+
+            ip4s = self.extractIP4s(text)
+            if ip4s is not None:
+                allIP4s.append(ip4s)
+
+            ip6s = self.extractIP6s(text)
+            if ip6s is not None:
+                allIP6s.append(ip6s)
+
+        # create events for emails, username and IPs
+        for email in allEmails:
+            email = str(email)
+            if eventData in email:
```

Comment:
This could result in a lot of false positives. Use `self.getTarget().matches()` instead.

Reply:
The email will never match the target exactly, so I've opted to use:

   if email.endsWith(self.getTarget()):

 Does that seem okay?

Reply:
Hmm that won't work, since getTarget() may also be an IP address or something else entirely. If you pass the domain of the email to `matches()`, it should be smart enough to handle this correctly.